### ISSUE TITLE
make: undefined reference to symbol 'dlclose@@GLIBC_2.2.5'

### ISSUE THREAD

**Issue by sincere-music** [palestrina/lily-q#4]
I tried compiling lily-q on a Ubuntu 16.04 live system, installed the dependencies (libasound2-dev and libreadline-dev; gcc was already installed, version 4:5.3.1-1ubuntu1). Make fails with this log:
[make-lily-q.log](https://github.com/palestrina/lily-q/files/1610368/make-lily-q.log)

**Comment by palestrina** [comment]
Hi Simon,

As far as I can tell, it looks like a problem with libdl (to do with dynamic linking). I am able to reproduce this problem:
`/usr/bin/ld: LuaSource/liblua.a(loadlib.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'`
...by leaving -ldl out of my Makefile. It’s a bit of a mystery why the problem happens with liblua.a, because I’m trying to include Lua statically. At any rate for me (Debian & Fedora), compilation seems to work with a “good” version of libdl.

Another alternative is to place -ldl later in the process (in the Makefile):
```
	gcc -ILuaSource -LLuaSource -o lq LilyQuick.c -lasound \
		-lpthread -llua -ldl -lm

```
From the tiny bit I know about linking (mostly that it is completely unintuitive), this might help. Don’t forget Makefiles need real tabs as indentation. Good luck!

**Comment by sincere-music** [comment]
Am 09.01.2018 um 03:12 schrieb palestrina:
> From the tiny bit I know about linking (mostly that it is completely 
> unintuitive), this might help. Don’t forget Makefiles need real tabs 
> as indentation. Good luck!

I’m sorry, but I don’t know anything about makefiles or the programming 
languages involved in this, so I’m afraid I’d have to rely on what is in 
the source package…

**Comment by palestrina** [comment]
Fair enough. I got around to releasing Version 1.0, which has the change in the Makefile. See if it works now.
